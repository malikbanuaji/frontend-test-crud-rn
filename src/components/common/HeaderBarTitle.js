import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {Spacing, Typography} from '../../styles';

export default function HeaderBarTitle({text}) {
  return <Text style={styles.headerBarTitle}>{text}</Text>;
}

const styles = StyleSheet.create({
  headerBarTitle: {
    ...Typography.headerBarTitle,
    flex: 1,
    marginLeft: Spacing.smaller,
  },
});
