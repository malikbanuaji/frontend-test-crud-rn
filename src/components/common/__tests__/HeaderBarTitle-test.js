import React from 'react';
import renderer from 'react-test-renderer';
import HeaderBarTitle from '../HeaderBarTitle';

test('renders correctly', () => {
  const tree = renderer.create(<HeaderBarTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props `text`', () => {
  const tree = renderer.create(<HeaderBarTitle text={'Hello'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
