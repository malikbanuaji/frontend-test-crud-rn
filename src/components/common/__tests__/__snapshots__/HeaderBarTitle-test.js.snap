// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    Object {
      "flex": 1,
      "fontFamily": "Rubik-Light",
      "fontSize": 24,
      "marginLeft": 8,
    }
  }
/>
`;

exports[`renders correctly with props \`text\` 1`] = `
<Text
  style={
    Object {
      "flex": 1,
      "fontFamily": "Rubik-Light",
      "fontSize": 24,
      "marginLeft": 8,
    }
  }
>
  Hello
</Text>
`;
